src/puppetlabs/puppetdb/mq_listener.clj

```diff
@@ -393,31 +331,72 @@
    handler-fn :- message-fn-schema]
   (swap! listener-atom conj [pred handler-fn]))
 
+(defn start-receiver
+  [connection endpoint discard-dir process-msg]
+  (let [sess (.createSession connection true Session/SESSION_TRANSACTED)
+        q (.createQueue sess endpoint)
+        consumer (.createConsumer sess q)
+        producer (.createProducer sess q)
+        send #(mq/commit-or-rollback sess
+                (.send producer (mq/to-jms-message sess %1 %2)))
+        handle (create-message-handler send discard-dir process-msg)]
+    (.setMessageListener
+     consumer
+     (reify MessageListener
+       (onMessage [this msg]
+         (try
+           (mq/commit-or-rollback sess (handle (mq/convert-jms-message msg)))
+           (catch Throwable ex
+             ;; Q: What do we really want to do here?
+             (log/error ex "message receive failed")
```

Comment:
I think all the Q: comments can be removed without code changes, unless there's something you think should be done in these cases aside from logging the exception.